qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -294,17 +301,53 @@ def _replace_node(self, dag, node, instr_map):
                     node.op.params, node.op.name, target_params, target_dag
                 )
             )
-
         if node.op.params:
-            # Convert target to circ and back to assign_parameters, since
-            # DAGCircuits won't have a ParameterTable.
-            target_circuit = dag_to_circuit(target_dag)
-
-            target_circuit.assign_parameters(
-                dict(zip_longest(target_params, node.op.params)), inplace=True
-            )
+            parameter_map = dict(zip_longest(target_params, node.op.params))
```

Comment:
The `zip_longest` confuses me a bit: isn't that redundant since the parameters are checked to have the same length above?

Reply:
Yeah, I guess it's not needed. I just used it because that's what the previous `assign_parameters` call did before I'll replace it with just a regular `zip()`, it'll be marginally faster.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/10707/commits/4ddf88876e13b8417c559e43c0a9ccafa76091ba

Reply:
I would argue for `zip(...., strict=True)` as the default for situations where different lengths imply a user or programmer error.  The only reason not to do this is that this check seems to be about 30% slower when there are few items. 

`zip(a, b, strict=False)` is semantically the same as `zip(a, b)`, but is 30% slower, presumably because the interpreter has to examine the last argument to determine it's not another iterable.

As the iterables get larger, I don't see a performance difference. When zipping lists of 1000 integers the fluctuations in timing at the cli are larger than the difference.